### ISSUE TITLE
Seems like the model outputs are disp image rather than depth image.

### ISSUE THREAD

**Issue by InFistLee** [DepthAnything/Depth-Anything-V2#40]
Thanks for the great work!

I noticed that the model outputs the disp image instead of depth image, how can I get the depth image from the model?

**Comment by InFistLee** [comment]
I have noticed that the code in `metric-depth` provides depth image, but I found the results were worse and many floater can be observed in the results. Since I don't need the absolute scaled depth and a relative one is okay, is there a way to easily convert the results from the original model to a relative depth map?

**Comment by Osierddr** [comment]
@InFistLee i have a same problem, do you have any idea recently?

**Comment by Liuhsinlun** [comment]
@InFistLee same issue here, i just can't find the code to generate the depth image like this :
![img](https://github.com/DepthAnything/Depth-Anything-V2/assets/143798011/4d194c54-6fb4-4a1e-804f-56c1ac296393)
![depth_map](https://github.com/DepthAnything/Depth-Anything-V2/assets/143798011/02d625d3-96e4-4b10-8cff-5e95b1300ec8)

Does anyone have the solution to this problem?

**Comment by AbedSoleymani** [comment]
@InFistLee To generate the depth map image (`depth.png`), and @Liuhsinlun to generate the coloured depth map (`depth_colored.png`) use this code:

```python
import cv2
import numpy as np
from PIL import Image
import matplotlib.pyplot as plt
import torch

from depth_anything_v2.dpt import DepthAnythingV2

DEVICE = 'cuda' if torch.cuda.is_available() else 'mps' if torch.backends.mps.is_available() else 'cpu'

model_configs = {
    'vits': {'encoder': 'vits', 'features': 64, 'out_channels': [48, 96, 192, 384]},
    'vitb': {'encoder': 'vitb', 'features': 128, 'out_channels': [96, 192, 384, 768]},
    'vitl': {'encoder': 'vitl', 'features': 256, 'out_channels': [256, 512, 1024, 1024]},
    'vitg': {'encoder': 'vitg', 'features': 384, 'out_channels': [1536, 1536, 1536, 1536]}
}

encoder = 'vitl' # or 'vits', 'vitb', 'vitg'

model = DepthAnythingV2(**model_configs[encoder])
model.load_state_dict(torch.load(f'checkpoints/depth_anything_v2_{encoder}.pth', map_location='cpu'))
model = model.to(DEVICE).eval()

raw_img = cv2.imread('path2RGB.png')
depth = model.infer_image(raw_img) # HxW raw depth map in numpy

depth_image = Image.fromarray(depth.astype(np.uint16))
depth_image.save('depth.png')


depth_normalized = (depth - depth.min()) / (depth.max() - depth.min()) * 255.0
depth_normalized = depth_normalized.astype(np.uint8)
colored_depth = (plt.cm.plasma(depth_normalized / 255.0)[:, :, :3] * 255).astype(np.uint8)

colored_depth_image = Image.fromarray(colored_depth)
colored_depth_image.save('depth_colored.png')
```

**Comment by Liuhsinlun** [comment]
@AbedSoleymani   Thank you sooo much!!!

**Comment by 520jz** [comment]
> @InFistLee To generate the depth map image (`depth.png`), and @Liuhsinlun to generate the coloured depth map (`depth_colored.png`) use this code:
> 
> ```python
> import cv2
> import numpy as np
> from PIL import Image
> import matplotlib.pyplot as plt
> import torch
> 
> from depth_anything_v2.dpt import DepthAnythingV2
> 
> DEVICE = 'cuda' if torch.cuda.is_available() else 'mps' if torch.backends.mps.is_available() else 'cpu'
> 
> model_configs = {
>     'vits': {'encoder': 'vits', 'features': 64, 'out_channels': [48, 96, 192, 384]},
>     'vitb': {'encoder': 'vitb', 'features': 128, 'out_channels': [96, 192, 384, 768]},
>     'vitl': {'encoder': 'vitl', 'features': 256, 'out_channels': [256, 512, 1024, 1024]},
>     'vitg': {'encoder': 'vitg', 'features': 384, 'out_channels': [1536, 1536, 1536, 1536]}
> }
> 
> encoder = 'vitl' # or 'vits', 'vitb', 'vitg'
> 
> model = DepthAnythingV2(**model_configs[encoder])
> model.load_state_dict(torch.load(f'checkpoints/depth_anything_v2_{encoder}.pth', map_location='cpu'))
> model = model.to(DEVICE).eval()
> 
> raw_img = cv2.imread('path2RGB.png')
> depth = model.infer_image(raw_img) # HxW raw depth map in numpy
> 
> depth_image = Image.fromarray(depth.astype(np.uint16))
> depth_image.save('depth.png')
> 
> 
> depth_normalized = (depth - depth.min()) / (depth.max() - depth.min()) * 255.0
> depth_normalized = depth_normalized.astype(np.uint8)
> colored_depth = (plt.cm.plasma(depth_normalized / 255.0)[:, :, :3] * 255).astype(np.uint8)
> 
> colored_depth_image = Image.fromarray(colored_depth)
> colored_depth_image.save('depth_colored.png')
> ```

Hello, I would like to ask a question, why is the size of the depth value opposite? For example, in the following picture, the closer the street is to the inside, the larger the value should be, but the value in the numpy array is smaller.
According to normal intuition, the greater the depth, the larger the value should be.

![image](https://github.com/user-attachments/assets/493dc9ce-57be-4b92-ac14-1d920015d303)
![image](https://github.com/user-attachments/assets/a09c6408-4fbf-4baf-aa95-e8ed5adb3c59)
![image](https://github.com/user-attachments/assets/91de8799-54f8-461b-8d24-79699006982f)

**Comment by Master-cai** [comment]
@520jz I have had the same question too. And I figured it out in their original paper sec 5.2
> Similarly, our models produce affine-invariant inverse depth

So the output is inverse depth, the real depth should be 1 / output.

**Comment by jsckdon** [comment]
May I ask what the unit of this depth value is, meters
